### PR TITLE
move lightning_fabric >> lightning/fabric

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

this will alow easier developing lightning in future with lower developer overhead

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @carmocca @akihironitta @justusschock @awaelchli

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -39,7 +39,7 @@
 /src/pytorch_lightning/core/module.py        @williamfalcon @tchaton @awaelchli @carmocca
 
 # Lightning Fabric
-/src/lightning_fabric                         @awaelchli @carmocca @justusschock
+/src/lightning/fabric                         @awaelchli @carmocca @justusschock
```
Comment (Borda) [inline .github/CODEOWNERS:42]:
```suggestion
/src/lightning_fabric                         @awaelchli @carmocca @justusschock
```
skip and do it at once :chipmunk:

**Review by akihironitta — APPROVED** [review]

**Comment by awaelchli** [comment]
You didn't update any imports in PL from lightning_fabric to lightning.fabric. How isn't it breaking in the CI?

**Comment by awaelchli** [comment]
Furthermore, the imports in tests of fabric weren't updated either. This is another indicator that something isn't right. The tests should run with the actual source code from this PR.

**Comment by Borda** [comment]
> This is another indicator that something isn't right. The tests should run with the actual source code from this PR.

hmm, I think I may see what makes the confusion; note that we install the package, not just use the local source, whhcih means that if installed with `PACKAGE_NAME=fabric` it still install the package as a standalone... :otter:

File: setup.py [hunk 2]
```diff
@@ -142,10 +142,9 @@ def _set_manifest_path(manifest_dir: str, aggregate: bool = False, mapping: Mapp
         package_to_install = _PACKAGE_MAPPING.get(_PACKAGE_NAME, "lightning")
         if package_to_install == "lightning":
             # merge all requirements files
-            assistant._load_aggregate_requirements(_PATH_REQUIRE, _FREEZE_REQUIREMENTS)
-        if package_to_install == "lightning" or _PACKAGE_NAME in _MIRROR_PACKAGE_REVERSED:  # install everything
-            # replace imports and copy the code
-            assistant.create_mirror_package(_PATH_SRC, _PACKAGE_MAPPING, reverse=_MIRROR_PACKAGE_REVERSED)
+            assistant._load_aggregate_requirements(_PATH_REQUIRE, _FREEZE_REQUIREMENTS)  # install everything
+        # replace imports and copy the code
+        assistant.create_mirror_package(_PATH_SRC, _PACKAGE_MAPPING, reverse=_MIRROR_PACKAGE_REVERSED)
```
Comment (Borda) [inline setup.py:147]: this is the place we create mirror fabric also for building PL package, as mention in https://github.com/Lightning-AI/lightning/pull/16589#issuecomment-1412007110

**Review by justusschock — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Pull request merged**